src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+  const result = [];
+
+  for (const item of actions) {
+    const stateCopy = { ...currentState };
+
+    if (item.type === 'addProperties') {
+      Object.assign(stateCopy, { ...item.extraData });
```

Comment:
The use of `Object.assign` with a spread operator on `item.extraData` is redundant. You can directly pass `item.extraData` to `Object.assign` without spreading it, as `Object.assign` already performs a shallow copy of the source objects.